### PR TITLE
feat: restructure developers category

### LINKED ISSUE #1494: docs: add all RaaS providers to menu
[Caldera](https://caldera.xyz/), [Conduit](https://conduit.xyz/), [AltLayer](https://altlayer.io/), and [Gelato](https://www.gelato.network/raas)

### LINKED ISSUE #1514: feat: restructure developers rollup category
```[tasklist]
### Tasks
- [x] make a button component for different frameworks, use on build modular page
- [x] Josh to update the 'Build Modular' section on the website to make it more streamlined and user-friendly, including creating separate sections/categories for different rollup types like EVM chains, sovereign rollups, and Celestia native rollups, adding logos or icons for easy identification of different rollup options, providing a clear path for users to quickly get to the deployment process, and removing unnecessary sections or content that might confuse users.
```

### PR DESCRIPTION
**Pull request by jcstein**
## Overview
WIP
Resolves #1514
Resolves #1494 
- [x] dark mode support (make dark mode background darker, highlight lighter)
- [x] finalize button styling

### Menu
Screenshot 2024-04-10 at 7 15 37 PM

### Buttons
Screenshot 2024-04-10 at 7 13 36 PM
Screenshot 2024-04-10 at 7 13 31 PM
Screenshot 2024-04-10 at 7 13 25 PM

## Summary by CodeRabbit

- **New Features**
  - Added new sections for "Astria" and "Sovereign" in the developers' sidebar navigation.
  - Introduced documentation links for "Astria" and deployment guides like "Deploy to Dusknet."
  - Added a new "Ethereum fallback mechanism" section under "Arbitrum."
  - Included a new "Other" section with a link to "Dymension."

- **Documentation**
  - Updated sidebar navigation links and structure under "Developers."
  - Expanded content on rollups and their benefits in the "Build Whatever" page.
  - Revised links across various documentation pages to reflect new navigation paths and content focus.

- **Bug Fixes**
  - Corrected navigation links and text references across multiple documentation pages to ensure consistency and accuracy.

### CODE REVIEW

File: .vitepress/config.ts [hunk 1]
```diff
@@ -257,7 +257,7 @@ function nav() {
         { text: "Learn", link: "/learn/how-celestia-works/overview" },
         { text: "Networks", link: "/nodes/participate" },
         { text: "Nodes", link: "/nodes/overview" },
-        { text: "Developers", link: "/developers/build-modular" },
+        { text: "Developers", link: "/developers/build-whatever" },
```
Comment (Bidon15) [inline .vitepress/config.ts:260]: Nice ❤️

File: .vitepress/config.ts [hunk 2]
```diff
@@ -664,33 +610,99 @@ function sidebarHome() {
...
-            { text: "Rollkit", link: "/developers/rollkit" },
             {
-              text: "Astria",
+              text: "Sovereign",
+              collapsed: true,
+              items: [
+              {
+                text: "Dymension",
+                link: "https://dymension.xyz/",
+              },
```
Comment (Bidon15) [inline .vitepress/config.ts:629]: They are not sovereign as they settle on dymension's hub. 
let's make the category others for now and put dymension there

File: .vitepress/config.ts [hunk 3]
```diff
@@ -664,33 +610,99 @@ function sidebarHome() {
...
+              { text: "Rollkit", link: "https://rollkit.dev" },
+              {
+                text: "Sovereign SDK",
+                link: "https://github.com/Sovereign-Labs/sovereign-sdk/tree/stable/examples/demo-rollup#demo-rollup",
+              },
+              {
+                text: "Vistara",
+                link: "https://docs.vistara.dev/",
```
Comment (Bidon15) [inline .vitepress/config.ts:637]: remove

**Review by Bidon15 — APPROVED** [review]

File: .vitepress/config.ts (hunk 3 above)
Comment (jcstein) [reply .vitepress/config.ts:637]: got it. will remove button on MD page too. however, it's listed on our site still on https://celestia.org/build/

File: .vitepress/config.ts (hunk 2 above)
Comment (jcstein) [reply .vitepress/config.ts:629]: got it. will make the change. what about astria?

File: .vitepress/config.ts (hunk 3 above)
Comment (jcstein) [reply .vitepress/config.ts]: https://github.com/celestiaorg/docs/pull/1515/commits/c6e53a38ee4fc6cef432c08547899907137e1169

File: .vitepress/config.ts (hunk 2 above)
Comment (jcstein) [reply .vitepress/config.ts]: https://github.com/celestiaorg/docs/pull/1515/commits/c6e53a38ee4fc6cef432c08547899907137e1169

File: .vitepress/config.ts (hunk 3 above)
Comment (jcstein) [reply .vitepress/config.ts]: fixed in https://github.com/celestiaorg/docs/pull/1515/commits/2436703eb9503de0273bf6fba2999b498abafc18

File: .vitepress/config.ts [hunk 4]
```diff
@@ -686,34 +610,124 @@ function sidebarHome() {
...
             },
-            { text: "Rollkit", link: "/developers/rollkit" },
             {
-              text: "Astria",
+              text: "Other",
+              collapsed: true,
+              items: [
+                {
+                  text: "Dymension",
+                  link: "https://dymension.xyz/",
```
Comment (jcstein) [inline .vitepress/config.ts:639]:
```suggestion
                  link: "https://docs.dymension.xyz/build/overview/",
```

Changing the link here @alex-beckett

File: developers/build-whatever.md [hunk 5]
```diff
@@ -2,16 +2,69 @@
...
+
+So, you’re ready to start experimenting and building on Celestia?
+Here are a few options that are currently available for developers.
+
+<div style="display: flex; flex-wrap: wrap; justify-content: center; align-items: center; grid-template-columns: repeat(auto-fill, minmax(150px, 1fr)); gap: 20px;">
+  <UrlImageButton url="/developers/intro-to-op-stack" imageSrc="/build/opstack.webp" text="OP Stack" notes="EVM" target="_self"/>
+  <UrlImageButton url="/developers/arbitrum-integration" imageSrc="/build/arbitrum.webp" text="Arbitrum Orbit" notes="EVM" target="_self"/>
+  <UrlImageButton url="/developers/build-whatever" imageSrc="/build/polygon.webp" text="Polygon CDK" target="_self" notes="EVM: Coming soon"/>
+  <UrlImageButton url="https://github.com/Sovereign-Labs/sovereign-sdk/tree/stable/examples/demo-rollup#demo-rollup" imageSrc="/build/sovereign.webp" text="Sovereign SDK" notes="Sovereign"/>
+  <UrlImageButton url="https://dymension.xyz/" imageSrc="/build/dymension.webp" text="Dymension"/>
```
Comment (jcstein) [inline developers/build-whatever.md:38]:
```suggestion
  <UrlImageButton url="https://docs.dymension.xyz/build/overview/" imageSrc="/build/dymension.webp" text="Dymension"/>
```

and here @alex-beckett

File: developers/integrate-celestia.md [hunk 6]
```diff
@@ -19,7 +19,7 @@ When getting started Celestia, we recommend checking out these resources first:
 - [Celestia's DA Layer](../learn/how-celestia-works/data-availability-layer.md)
 - [Learn modular](https://celestia.org/learn.md)
 - [Overview to running nodes on Celestia](../nodes/overview.md)
-- [Build Moduar](./build-modular.md)
+- [Build Moduar](./build-whatever.md)
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/integrate-celestia.md]: https://github.com/celestiaorg/docs/pull/1515/commits/bd442ed990b5b40f3c5a9c107580d8ac658f1d01

File: developers/build-whatever.md [hunk 7]
```diff
@@ -2,16 +2,69 @@
...
+It’s layer 1s job to order those transactions and check that they are
+available, at minimum.
+
+## Quickstart - Building on Celestia
+
+### Choose a framework
+
+<script setup>
+import UrlImageButton from '../.vitepress/components/UrlImageButton.vue';
+</script>
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/build-whatever.md:28]: this doesn't work

File: developers/build-whatever.md [hunk 8]
```diff
@@ -2,16 +2,69 @@
...
+Here are a few options that are currently available for developers.
+
+<div style="display: flex; flex-wrap: wrap; justify-content: center; align-items: center; grid-template-columns: repeat(auto-fill, minmax(150px, 1fr)); gap: 20px;">
+  <UrlImageButton url="/developers/intro-to-op-stack" imageSrc="/build/opstack.webp" text="OP Stack" notes="EVM" target="_self"/>
+  <UrlImageButton url="/developers/arbitrum-integration" imageSrc="/build/arbitrum.webp" text="Arbitrum Orbit" notes="EVM" target="_self"/>
+  <UrlImageButton url="/developers/build-whatever" imageSrc="/build/polygon.webp" text="Polygon CDK" target="_self" notes="EVM: Coming soon"/>
+  <UrlImageButton url="https://github.com/Sovereign-Labs/sovereign-sdk/tree/stable/examples/demo-rollup#demo-rollup" imageSrc="/build/sovereign.webp" text="Sovereign SDK" notes="Sovereign"/>
+  <UrlImageButton url="https://docs.dymension.xyz/build/overview/" imageSrc="/build/dymension.webp" text="Dymension"/>
+  <UrlImageButton url="https://rollkit.dev" imageSrc="/build/rollkit.webp" text="Rollkit" notes="Sovereign"/>
+</div>
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/build-whatever.md]: https://github.com/celestiaorg/docs/pull/1515/commits/24291b2d2fae6938d1d91190f8db409126c404aa

File: developers/integrate-celestia.md [hunk 9]
```diff
@@ -19,7 +19,7 @@ When getting started Celestia, we recommend checking out these resources first:
 - [Celestia's DA Layer](../learn/how-celestia-works/data-availability-layer.md)
 - [Learn modular](https://celestia.org/learn.md)
 - [Overview to running nodes on Celestia](../nodes/overview.md)
-- [Build Moduar](./build-modular.md)
+- [Build Modular](./build-whatever.md)
```
[the remark this answers is not in the document]
Comment (jcstein) [reply developers/integrate-celestia.md]: https://github.com/celestiaorg/docs/pull/1515/commits/244523014fe4c6e70b0deb9e287b44e8a7208176

**Pull request merged**